perf/ListPool.Benchmarks/EnumerableToListPoolBenchmark.cs

```diff
@@ -27,7 +27,7 @@ public void IterationSetup()
                 items[i] = 1;
             }
 
-            _items = items.Select(i => i);
+            _items = items;
```

Comment:
problem was here
reason: select it is lazy operation, it was called for ListPool first and it is added more time to it, and after that when select was called benchmark for list started :D

Reply:
I don't think this is the reason. You can enumerate multiple times an enumerable and it won't change its state. ToList will allocate "materialized" a list in each call. Additionally, This method is running each iteration.

You can test it also by commenting one test and running then separately.

Looking at CoreFx repository, I see that they have a type IIListProvider<TSource> with some optimized path. I think we should give a look to it.

https://github.com/dotnet/corefx/blob/8750960d3fafa46a9b838c351e995a01fa8b599f/src/System.Linq/src/System/Linq/ToCollection.cs#L23

Reply:
It is just internal interface which implemented for each step (Select/Grouping) to make it lazy.
It is combine previous steps and new one and call it in one time iteration instead of call it each time.
When it call ToList in case when it is Linq it is just calling after Selection IListProvider implementation, in part of our implementation we not possible to work with lazy implementation which used in Linq, in this case when it is tried to get inumerator first it is return which used in linq and call lazy operations in iterator, just after calling ToListPool(), which created new ListPool. For LINQ it is optimized and when we call ToList it is called implementation of iterator and call lazy operations in one iteration.
https://github.com/dotnet/corefx/blob/8750960d3fafa46a9b838c351e995a01fa8b599f/src/System.Linq/src/System/Linq/Grouping.SpeedOpt.cs#L9

Reply:
I want to check just one more thing, will update pr

Reply:
Yes, you are right. LINQ has deferred operations and knows the original type in some cases, so it can perform faster operation for example in Array that we can do because their return type is always IEnumerable.

Instead of refactoring this test to use Enumerable, I think we should add more tests with List, Array, and Enumerable, to make clear the difference. And also think in a better way how we can name our benchmarks.